File: app/src/main/java/com/bsas/androiddevs/manejoerrores/repository/api/MoviesEndpoints.java
package com.bsas.androiddevs.manejoerrores.repository.api;

import com.bsas.androiddevs.manejoerrores.repository.api.dto.GetMoviesResponse;

import retrofit2.Call;
import retrofit2.http.GET;
import retrofit2.http.Query;

public interface MoviesEndpoints {

    @GET
    Call<GetMoviesResponse> getMovies(@Query("apikey") String apiKey,
                                      @Query("s") String query,
                                      @Query("type") String movieType);

}

File: app/src/main/java/com/bsas/androiddevs/manejoerrores/repository/api/MoviesEndpoints.kt
package com.bsas.androiddevs.manejoerrores.repository.api

import com.bsas.androiddevs.manejoerrores.repository.api.dto.GetMoviesResponse
import retrofit2.Response
import retrofit2.http.GET
import retrofit2.http.Query

interface MoviesEndpoints {

    @GET("/")
    suspend fun getMovies(@Query("s") query: String,
                          @Query("type") movieType: String = MoviesApiClient.MOVIE_TYPE_QUERY,
                          @Query("apikey") apiKey: String = MoviesApiClient.API_KEY): Response<GetMoviesResponse>


}